README.Rmd

```diff
@@ -74,6 +77,18 @@ template:
   trailing_slash_redirect: true
 ```
 
+If your site should be added to a subdomain such as 
+`tidyverse.org`, `r-lib.org`, `tidymodels.org`, then follow these steps:
+
+```md
+* [ ] Submit PR [here](https://github.com/rstudio/aws-main/tree/main/zones) \\
+      adding your site to the appropriate domain (eg., r-lib, tidyverse, tidymodels)
+* [ ] Add `url: https://{pkgname}.{domain}.org` to `_pkgdown.yml`
+* [ ] Edit the URL field in DESCRIPTION and re-document.
+* [ ] Set url in GitHub Settings > Pages > Custom Domain
+* [ ] Edit URL in the settings in the "About" section of the package's GitHub page
```

Comment:
I feel like there's some sequence where you can set the subdomain and then have this other stuff happen? Maybe @jennybc knows?

Reply:
In the "About" Settings there is a checkbox to use your GitHub pages website, which makes that step less manual, but other than that I'm not sure

Reply:
I thought `use_pkgdown_github_pages()` will do most of this for you, but I don't remember how you give it the URL to start the whole process.

Reply:
Oh maybe it's this comment in the docs:

> Packages owned by certain GitHub organizations (tidyverse, r-lib, and tidymodels) get some special treatment, in terms of anticipating the (eventual) site URL and the use of a pkgdown template.

Reply:
Ok, yeah I see that in `tidyverse_url()` now. It also looks like `use_github_pages()` pulls the CNAME from the repo and then `use_pkgdown_url()` uses that to set the GitHub homepage url, populate the url field in `_pkgdown.yml`, and add the URL in the DESCRIPTION. So perhaps the CNAME needs to be configured and the custom domain needs to be set manually in the repo settings, then call `use_pkgdown_github_pages()` will do the rest?

Reply:
I'm pretty sure that `use_pkgdown_github_pages()` just does all of this and is probably harmless to call in a repo with existing GitHub pages / pkgdown setup.

I think at this point, the best thing to do is try it to double check me, in some suitable repo.

Reply:
`use_pkgdown_github_pages()` will overwrite a customized `_pkgdown.yml`, but as discussed the diff can be used to easily keep the necessary old bits and integrate them with the new.

Reply:
I just did it with `usethis` and I think almost all is done for you except setting the custom domain, so I think that needs to be done manually before calling `use_pkgdown_github_pages()`:

```
> use_pkgdown_github_pages()
✔ Writing '_pkgdown.yml'
• Modify '_pkgdown.yml'
✔ GitHub Pages is publishing from:
• Custom domain: 'usethis.r-lib.org'
• URL: 'https://usethis.r-lib.org/'
• Branch: 'gh-pages'
• Path: '/'
✔ Saving 'r-lib/actions/examples/pkgdown.yaml@v2' to '.github/workflows/pkgdown.yaml'
• Learn more at <https://github.com/r-lib/actions/blob/v2/examples/README.md>.
✔ Recording 'https://usethis.r-lib.org/' as site's url in '_pkgdown.yml'
✔ Adding 'https://usethis.r-lib.org/' to URL
• Run `devtools::document()` to update package-level documentation.
✔ Setting 'https://usethis.r-lib.org/' as homepage of GitHub repo 'r-lib/usethis'
```